docs/template_plugin/tests/functional/op_reference/pad.cpp

```diff
@@ -116,6 +116,64 @@ TEST_P(ReferencePadTestParamsOk, CompareWithRefs) {
     EXPECT_NO_THROW(Exec());
 }
 
+class ReferencePadTestNonConstPadsBeginPadsEnd : public ReferencePadTest {
+public:
+    void SetUp() override {
+        SKIP_IF_CURRENT_TEST_IS_DISABLED();
+        auto params = GetParam();
+        function = CreateFunction(params);
+        inputData = {params.inputData.data, params.padsBegin.data, params.padsEnd.data};
+        refOutData = {params.expectedOutput.data};
+    }
+
+private:
+    static std::shared_ptr<Function> CreateFunction(const PadParams& params) {
+        const auto data = std::make_shared<op::v0::Parameter>(params.inputData.type,
+                                                              params.inputData.shape);
+        const auto padsBegin = std::make_shared<op::v0::Parameter>(params.padsBegin.type,
+                                                                   params.padsBegin.shape);
+        const auto padsEnd = std::make_shared<op::v0::Parameter>(params.padsEnd.type,
+                                                                 params.padsEnd.shape);
+        const auto f = [&] {
+            if (params.useConstValue) {
+                // pad_value should be used only in CONSTANT mode
+                const auto padVal = op::v0::Constant::create(params.constantValue.type,
```

Comment:
In general I think pad value also can be non-constant. Can we also check this case?

Reply:
It works fine. New test will be updated with PadVal as Parameter in next PR